packages/transducers/src/iter/norm-range.ts

```diff
@@ -21,3 +21,37 @@ export function* normRange(
         }
     }
 }
+
+export function* normRange2d(
+    nx: number,
+    ny: number,
+    includeLastX: boolean = true,
+    includeLastY: boolean = true,
+) { 
+    const rx = normRange(nx, includeLastX);
+    for (let y of normRange(ny, includeLastY)) {
+        for (let x of rx) {
+            yield [x, y];
+        }
+    }
+}
+
+export function* normRange3d(
+    nx: number,
+    ny: number,
+    nz: number,
+    includeLastX: boolean = true,
+    includeLastY: boolean = true,
+    includeLastZ: boolean = true,
+) { 
+    const rx = normRange(nx, includeLastX);
+    const ry = normRange(ny, includeLastY);
```

Comment:
Same issue as in `normRange2d` (caching inner results, i.e. `rx`, `ry`) - or - we could also express `normRange3d()` as a concatenation of `normRange2d()` XY slices:

```ts
// normal function now!
export function normRange3d(
    nx: number,
    ny: number,
    nz: number,
    includeLastX: boolean = true,
    includeLastY: boolean = true,
    includeLastZ: boolean = true,
) {
  const sliceXY = [...normRange2d(x, y, includeLastX, includeLastY)];
  return mapcat((z) => map((xy) => [...xy, z], sliceXY), normRange(nz, includeLastZ))
}
```